File: javascript/bubble.js
var swap = require('./util').swap

// bubble implements bubble sort algorithm. The function takes an array.
function bubble(a) {
	for(var i = 0; i < a.length; i++) {
		var swapped = false
		for(var j = a.length-1; j > i; j--) {
			if(a[j] < a[j-1]) {
				swap(a, j, j-1)
				swapped = true
			}
		}
		if(!swapped) {
			break
		}
	}
}

module.exports = bubble

File: python/bubble.py
import util

# sort implements bubble sort algorithm. The function takes an array.
def sort(a):
	for i in range(len(a)):
		swapped = False
		for j in range(len(a)-1, i, -1):
			if a[j] < a[j-1]:
				util.swap(a, j, j-1)
				swapped = True
		if not swapped:
			break
